### PR TITLE
Adds the Shitty Still

### PR DESCRIPTION
**Pull request by Carbadox**
## About the PR 
Adds the shitty still to the game and most maps.
Works similar to the regular still, however brewing anything in the shitty still will also add a random base reagent (like the ones available in a ChemDispenser) to the mix.
The shitty still can be usually found in the more dingy parts of the station and/or at the Space Diner and John Bill's bus.

## Why's this needed? 
To expand the "homelessness" gameplay mechanics. 

## Changelog
```
(u)Carbadox:
(*)Added the shitty still! This amalgamation of scrap parts is used just like a regular still, however, it will also a random base chemical to the final product. The shitty still can be found at the Space Diner and John Bill's bus or around the "undesirable" parts of the station.
```

### CODE REVIEW

File: code/modules/chemistry/tools/dispensers.dm [hunk 1]
```diff
@@ -562,6 +562,38 @@
...
 
+/obj/reagent_dispensers/still/shittystill
+	name = "shitty still"
+	desc = "Looks like it some sort of makeshift distillery, made out of salvaged parts."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "shittystillFIN"
+	amount_per_transfer_from_this = 20
+	var/list/random_reagents_list = list("barium","flourine","lithium","magnesium","mercury",\
+"oxygen","phosphorus","plasma","potassium","radium","silver","sugar","hydrogen","water",\
+"carbon","chlorine","sulfur","calcium","iron","ethanol")
```
Comment (Michaellaneous) [inline code/modules/chemistry/tools/dispensers.dm:573]: Maybe add some fun reagents as well, ala discount dan; with a lesser chance and smaller amount?

**Review by Michaellaneous — CHANGES_REQUESTED** [review]

File: code/modules/chemistry/tools/dispensers.dm (hunk 1 above)
Comment (Carbadox) [reply code/modules/chemistry/tools/dispensers.dm:573]: Definitely a planned feature. Will probably add something in a separate PR alongside some additional stuff

Comment (Michaellaneous) [reply code/modules/chemistry/tools/dispensers.dm:573]: Makes sense!

**Comment by geneva-breaker** [comment]
This is a little rough. 
firstly "clarion, cog1, cog2 and oshan" != most maps. It's also poor form to remove objects from a map arbitrarily to replace them with your thing, and there doesnt seem to be much of a pattern as to what this is replacing. A punching bag, a crate and some evidence boxes? 

second, this replaces an object in an adventurezone, one that is kind of a theme-setting piece, but also just a weird thing to do imo. Also, if you're putting on on the Bus, why put another one in an area reachable by Bus? 

third, it would be nice if the still offered something more than just "theres a random innocuous chem taking up space in your container now". for how (inconsistently) available it is, I would expect a unique function or something more to differentiate it. 

TLDR: make it something unique, at least situationally desireable, and place it somewhere reachable but not in triplicate.

**Comment by Gerhazo** [comment]
Of note to consider: this removes and/or replaces some other things from the maps to add this thing, so isn't just an addition.

The icon state's name, "shittystillFIN", the "FIN" part seems more of a working title and probably shouldn't be there.

And of personal opinion: magically creating reagents feels weird or at least unintuitive of a mechanic to me. Especially in a mechanic (the still itself) that specifically takes into account the kind of produce being loaded in.
+ it feels like in most cases it'd lead to people randomly blowing themselves up, potentially damaging the station while at it.

some of these points were brought up already by warc as I was writing this I have come to notice, but am leaving them as my own thoughts anyways. I don't think this is unique enough or fills any niche to fit in the game, not to distract from its value of having hopefully been a learning experience having myself helped out occasionally with some parts of the code.

**Comment by pali6** [comment]
Creating random reagents is not intuitive but it's desirable for ghetto staff assistant chemistry in deep maint since you can't really get most of the base reagents without a chem dispenser currently.

**Comment by Michaellaneous** [comment]
Isn't creating random reagents basically the concept behind discount dans? Except for the list being too generic and the afformentioned location issues, I don't see a big problem with this.

**Comment by Carbadox** [comment]
I'll try and summarise my reponse in to one comment.

For a start, I'll admit definetely should have provided a better description for the "Why's this needed" section. I have a pretty bad habit where I often forget that I haven't explained my entire thought proccess behind something and expect other people to know what I mean/talking about, so yeah.

When I first started working on this, I originally wanted the shitty still to be just a way of sourcing your own homemade hobo-wine (the stuff thats found around stations, usually with wine, ethanol and something random like glitter). That originally also involved being able to actually construct the still by hand with station parts. However, after being told and finding out first hand that item construction has no proper framework I decided to abandon that idea and soon came around to what the shitty still is currently. Which is, a source for all base ChemDispenser reagents. That's why the sprites it called "shittystillFIN" since it was originally going to be the final result of the construction (and I hope to return and add that if said framwork ever does get added).

The "logical" explaination as to why these chems come up while brewing is that because of the bootleg machinery used for brewing, the chems are impurities that come about from the parts, imagine rusty pipes, used plasma tanks, etc. The risk of blowing up is intentional because distilleries do actually blow up in real life if there's an error in the procedures, so I think it's expected to happen if the proccess is happening in some sort of ghetto set-up. The chance of explosions is also avoidable if you're wary and pour out anything that might combine into a dangerous mix.

Regarding the map stuff, I'll see what I can do though its going to be a bit hard because the whole reason why the map implementation doesnt work is because I'm stuck with my dingy laptop again and any mapping program I try to use either refuses to work because it's a x32 system or the program just crashes. The only other way I could possibly add anything in accurately was by looking at the maps and straight up rewriting the map in VS Code. I'll try and revert those changes and find a better spot.

I put the still on maps but not on all of them, since I wanted them to be in locations that fit, EG Cogmap 2's crew quarters (the one that looks like a drug den) and areas like that aren't on every map. Which I why I also put one on the Space Diner for reliability's sake and since the shitty grill is already on Johns Bus, I feel like its straight up a missed opportunity to not have the shitty still alongside it.

Anyways, that's about the best reply I can give. Sorry if it's clunky, Im available to giving better answers on Discord if that's the preffered method.

**Comment by Gerhazo** [comment]
I'm slightly iffy because I feel that with all the clunkiness that this would involve: finding things to stuff in the still in the first place, then generating the random reagents, at which point there isn't a clean way to divide them, so you'd either be very wastefully isolating single ones or hoping that you don't blow up from having multiple, in addition to in general being likely to trigger other reactions than the one you're looking for, you needing to get a hand on a chemmaster to isolate out the things you need, a reagent scanner to conveniently check what you have and/or other chemistry equipment... 

It feels like you'd save time and be better off just either breaking into chemistry or breaking into mechanics to get supplies to build your own, as this feels for me equivalent to mashing random buttons at the dispenser, which we know generally doesn't work out well and trying to separate out the things you want will be very slow and inefficient.

That being said, those are just speculations on how I think it'd work out in practice, I may or may not be wrong in my assumptions. As well as those being just my personal opinions.

**Comment by zjdtmkhzt** [comment]
For a different way of doing things, what if certain things you put into the still gave certain basic chems, like for instance, brewing up chocolate will give some sugar or brewing up popsicles will give nitrogen or whatever.
Doesn't necessarily have to make any sense, but just so it's predictable and can be used for actual chemistry without relying on random chance.
Of course, it would be quite a bit of work to make a sensible list of conversions for this, but I think it would really make hobochem using this a lot more viable.

Also, I strongly agree that removing things from maps without consideration is not a good thing to do, and I think that the best locations for this would be simply putting one on the bus and making it constructable as well, even if the construction code is a bit janky.
